src/wp-includes/script-loader.php

```diff
@@ -2267,38 +2267,51 @@ function wp_enqueue_global_styles() {
 		return;
 	}
 
+	$separate_assets = wp_should_load_separate_core_block_assets();
+
+	/*
+	 * Global styles should be printed in the head when loading all styles combined.
+	 * The footer should only be used to print global styles for classic themes with separate core assets enabled.
+	 *
+	 * See https://core.trac.wordpress.org/ticket/53494.
+	 */
+	if ( ! $separate_assets && doing_action( 'wp_footer' ) || $separate_assets && doing_action( 'wp_enqueue_scripts' ) ) {
+		return;
+	}
+
 	$can_use_cache = (
 		( ! defined( 'WP_DEBUG' ) || ! WP_DEBUG ) &&
 		( ! defined( 'SCRIPT_DEBUG' ) || ! SCRIPT_DEBUG ) &&
 		( ! defined( 'REST_REQUEST' ) || ! REST_REQUEST ) &&
 		! is_admin()
 	);
 
-	$stylesheet = null;
-	if ( $can_use_cache ) {
-		$cache = get_transient( 'global_styles' );
-		if ( $cache ) {
-			$stylesheet = $cache;
+	if ( doing_action( 'wp_enqueue_scripts' ) ) {
```

Comment:
We need to remove this conditional, correct? Otherwise, the styles won't be enqueued when the action is `wp_footer`.

Reply:
Ah yes. Good catch.